src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTransformer.java

```diff
@@ -0,0 +1,253 @@
+
+    view.retryCodesDefinitions(generateRetryCodesDefinitions(context));
+    view.retryParamsDefinitions(generateRetryParamsDefinitions(context));
+
+    view.pathTemplates(pathTemplateTransformer.generatePathTemplates(context));
+    view.pathTemplateGetters(pathTemplateTransformer.generatePathTemplateGetterFunctions(context));
+    view.callSettings(apiCallableTransformer.generateCallSettings(context));
+    view.apiMethods(generateApiMethods(context));
+
+    ServiceConfig serviceConfig = new ServiceConfig();
+    view.serviceAddress(serviceConfig.getServiceAddress(service));
+    view.servicePort(serviceConfig.getServicePort());
+    view.authScopes(serviceConfig.getAuthScopes(service));
+
+    return view.build();
+  }
+
+  private ModelTypeTable createTypeTable() {
+    return new ModelTypeTable(new GoTypeTable(), new GoModelTypeNameConverter());
+  }
+
+  /**
+   * Returns the doc comments of Go for the proto elements.
+   */
+  public static List<String> comments(ProtoElement element) {
+    if (!element.hasAttribute(ElementDocumentationAttribute.KEY)) {
+      return ImmutableList.<String>of("");
+    }
+    return new GoContextCommon().getCommentLines(DocumentationUtil.getScopedDescription(element));
+  }
+
+  private List<StaticLangApiMethodView> generateApiMethods(SurfaceTransformerContext context) {
+    List<StaticLangApiMethodView> apiMethods = new ArrayList<>();
+
+    for (Method method : context.getNonStreamingMethods()) {
+      MethodConfig methodConfig = context.getMethodConfig(method);
+      MethodTransformerContext methodContext = context.asMethodContext(method);
+
+      if (methodConfig.isPageStreaming()) {
+        apiMethods.add(apiMethodTransformer.generatePagedRequestObjectMethod(methodContext));
+      } else {
+        apiMethods.add(apiMethodTransformer.generateRequestObjectMethod(methodContext));
+      }
+    }
+
+    return apiMethods;
+  }
+
+  /**
+   * Returns the Go package name.
+   *
+   * Retrieved by splitting the package string on `/` and returning the second to last string if
+   * possible, and otherwise the last string in the resulting array.
+   *
+   * TODO(saicheems): Figure out how to reliably get the intended value...
+   */
+  private static String getPackageName(String configPackageName) {
+    String split[] = configPackageName.split("/");
+    if (split.length - 2 >= 0) {
+      return split[split.length - 2];
+    }
+    return split[split.length - 1];
+  }
+
+  /**
+   * Returns the service's client name prefix.
+   */
+  private String getClientPrefix(Model model, Interface service, String packageName) {
+    String name = getReducedServiceName(service);
+    // If there's only one service, or the service name matches the package name, don't prefix with
+    // the service name.
+    if (model.getSymbolTable().getInterfaces().size() == 1 || name.equals(packageName)) {
+      return "";
+    }
+    return LanguageUtil.lowerUnderscoreToUpperCamel(name);
```

Comment:
Don't make any calls to LanguageUtil from any of the viewmodel/transformer code; its functionality is completely superseded by the Name class. Also, don't make any explicit casing calls in transformers; go through SurfaceNamer.